### PR TITLE
Added EJ RANGE 24 paper type and fixed a bug

### PR DESCRIPTION
**Pull request by LyricDev**
Great tool, many thanks for putting it together!

I am using these labels I bought on amazon and wanted to add a layout for it:
https://www.amazon.co.uk/gp/product/B07N5VBBPC

After creating a layout with the appropriate measurements I could see that the top row was above the edge of the canvas. Looking at the code, it became clear there is a layout issue here. When creating StickerRect, the bottom coord is calculated from the row number. As this is zero-indexed, this number is 0 for the top row, putting the BOTTOM of the top row at the margin point. This is where the TOP of the top row should be. Simply adding 1 to the index resolves this.

I am unclear how the existing layouts for your paper types print correctly (I don't have the measurements to check them). I suspect the top margin value has been enlarged to make them work? Anyway, it is confusing that the top-margin layout entry currently controls where the BOTTOM of the top row should go. This makes it more difficult for others who wish to add their own paper types.

### CODE REVIEW

File: LabelGenerator.py
Comment (Finomnis) [inline LabelGenerator.py:106]: I can't merge this as-is, as it breaks current generated files. I'll fix and rebase this PR afterwards.

**Comment by Finomnis** [comment]
Hey!
Thanks for your merge request.
Yes, I think the variables are just labeled wrong throughout a couple of refactorings.
I'll check and fix it, and then I'll rebase your PR on top of it.
Let me come back to you in a bit.

File: LabelGenerator.py
Comment (Finomnis) [reply LabelGenerator.py:106]: Fixed in upstream

**Comment by Finomnis** [comment]
I fixed it in the main branch. `margin_top` now actually means the distance from the top of the page to the first sticker.

Please rebase and adjust your numbers, your fix to the formula was incorrect, so your stickers might actually move with my fix.
Once you have done that and verified that it prints correctly at your page, I'll gladly accept your contribution.

Thank you very much for giving to the Open Source community :)

File: LabelGenerator.py
Comment (LyricDev) [reply LabelGenerator.py:106]: You are absolutely correct. For my paper type sticker_height = vertical_stride so I failed to notice the distinction.

**Comment by LyricDev** [comment]
Thanks for your review. I was mainly concerned about the impact on your existing layouts - have no measurements to check for myself if they need adjusting after the change. The current code works fine for the paper type I added.

I'm very new to git and open source contribution so forgive my ignorance of the process. Please let me know what I should do next (if anything). I'm keen to get better at contributing!

**Comment by Finomnis** [comment]
I wanted to avoid a bunch of back-and-forth so I just pushed the formatting fixes myself, I hope you don't mind. If more people participate I should set up a CI that checks for things like that.

**Comment by Finomnis** [comment]
Don't worry, you are doing great :) Thanks for contributing.

Normally you would now rebase to master, then I would go over your changes, the CI would run (which doesn't exist yet) and I would accept the change or add further remarks.

But as you already said that your paper works on your machine I'll rebase myself and fix the merge conflicts.

**Comment by LyricDev** [comment]
Thats absolutely fine. I hacked the code a bit locally to get some coloured label backgrounds for major values - I want them to act as highly visible dividers! If I have time at some point I may tidy it up and send another PR with an enhancement. 

Thanks again for the tool!

**Comment by Finomnis** [comment]
@LyricDev I rebased your change to master. I now need two things from you:

- Look over the 'changed files' tab and verify that the changes are as you intended, after the rebase
- Confirm that the stickers are still generated correctly for your paper

Once you can confirm, you can merge this change.

**Review by Finomnis — APPROVED** [review]

**Comment by LyricDev** [comment]
All looks good. Layout for EJ papers works correctly after rebase. 

Perhaps also add EJ Range 24 to the list of supported papers in the readme?

**Comment by Finomnis** [comment]
> Perhaps also add EJ Range 24 to the list of supported papers in the readme?

Is it still meant for 7x10cm (3"x4") bags? Or do you use a different size

**Comment by LyricDev** [comment]
I'm actually using 8X12cm bags though the EJ labels would still fit comfortably on 7x10.

**Comment by LyricDev** [comment]
Honestly I would just have bought the supported Avery, but its not a popular or easy to get hold of brand here in the UK!

**Comment by Finomnis** [comment]
After a little thinking I think I'll leave it out for now. The labels are a different ratio than originally intended and so the content is kinda stretched and it has a little too much whitespace, overall the design is not completely as intended. I don't have a problem having it as a selectable option in the script, but it deviates too much from the original to advertise official support imo.

**Comment by LyricDev** [comment]
Sure thing

**Comment by Finomnis** [comment]
@LyricDev Btw, I'm from Germany and have the same problem. But it's quite easy to find the Avery 5260 on ebay, I just looked at ebay.co.uk, and they seem to have them as well.

**Comment by LyricDev** [comment]
Nice one! I just checked and all the Avery 5260 I can find on ebay.co.uk have to shipped from the US!! Anyway, I'm happy with the end result, makes a big improvement over my resistor file which keeps spilling out random pieces.

**Comment by Finomnis** [comment]
@LyricDev I just found the Avery 8160 on amazon.co.uk, which have the same sticker size as the 5260. For future reference, in case someone reads this :)

**Comment by Finomnis** [comment]
But you are right, if you found a solution that works for you, that's the main thing :) have fun with it!

**Comment by LyricDev** [comment]
Thanks again!

**Pull request merged**
